dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/imports/ImportDDIServiceBean.java

```diff
@@ -281,30 +196,54 @@ private void processCodeBook(ImportType importType, XMLStreamReader xmlr, Datase
                     // metadata using the direct (non-harvesting) import API.
                     // EMK TODO: add this back in for ImportType.NEW
                     //processFileDscr(xmlr, datasetDTO, filesMap);
-                }
+                //}
 
             } else if (event == XMLStreamConstants.END_ELEMENT) {
                 if (xmlr.getLocalName().equals("codeBook")) {
+                    applyTermsOfUseToFiles(datasetDTO, termsOfUseDataHolder);
                     return;
                 }
             }
         }
     }
 
+    /**
+     * Applies terms of use into all of the files in dataset.
+     * Do note that if there is any data in {@link TermsOfUseDataHolder#getTermsOfUse()},
+     * it will be interpreted by this method as a license. It is not necessarily
+     * the truth. And we let further processing (parsing {@link DatasetDTO} to entity model)
+     * to decide whether some license should be assigned or not.
+     */
+    private void applyTermsOfUseToFiles(DatasetDTO datasetDTO, TermsOfUseDataHolder termsOfUseDataHolder) {
+        for (FileDTO file: datasetDTO.getDatasetVersion().getFiles()) {
+            String termsOfUse = termsOfUseDataHolder.getTermsOfUse();
+            
+            if (StringUtils.isNotBlank(termsOfUse)) {
+
+                if (termsOfUse.equals(TERMS_OF_USE_CC0_LICENSE)) {
+                    file.setTermsOfUseType(TermsOfUseType.LICENSE_BASED.toString());
+                    file.setLicenseName(License.CCO_LICENSE_NAME);
+                } else {
+                    file.setTermsOfUseType(TermsOfUseType.LICENSE_BASED.toString());
+                    file.setLicenseName(termsOfUseDataHolder.getTermsOfUse());
+                }
+            } else {
+                file.setTermsOfUseType(TermsOfUseType.TERMS_UNKNOWN.toString());
+            }
+        }
+    }
+
     private void processDocDscr(XMLStreamReader xmlr, DatasetDTO datasetDTO) throws XMLStreamException {
         for (int event = xmlr.next(); event != XMLStreamConstants.END_DOCUMENT; event = xmlr.next()) {
             if (event == XMLStreamConstants.START_ELEMENT) {
 
-                if (xmlr.getLocalName().equals("IDNo") && StringUtil.isEmpty(datasetDTO.getIdentifier())) {
+                if (xmlr.getLocalName().equals("IDNo") && StringUtils.isBlank(datasetDTO.getIdentifier())) {
```

Comment:
Actually from my standpoint either we will do it in all occurences in this class or keep it like this.  I could change it. It is not so much of a problem. But we got to keep in mind that it will not save us from npe's. Xml reader will throw illegalstateexception if we would try to execute getlocalname when the cursor is currently on an element that does not have it. So what do you think?

Reply:
As we'll nevertheless have an exception in the case you've described, we could leave all these `equals` as they currently are.